### PR TITLE
fix: Removed misleading watch-postGoal warning + fixed rolling update in DockerImageWatcher

### PR DESCRIPTION
**Pull request by manusa**
## Description

fix: Removed misleading watch-postGoal warning + fixed rolling update in DockerImageWatcher

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [ ] ~~I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly~~
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [x] I tested my code in Kubernetes
 - [ ] ~~I tested my code in OpenShift~~

### CODE REVIEW

File: jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java [hunk 1]
```diff
@@ -145,38 +145,33 @@ private void updateImageName(KubernetesClient kubernetes, String namespace, HasM
         if (entity instanceof Deployment) {
             Deployment resource = (Deployment) entity;
             DeploymentSpec spec = resource.getSpec();
-            if (spec != null) {
-                if (updateImageName(entity, spec.getTemplate(), imagePrefix, imageName)) {
-                    kubernetes.apps().deployments().inNamespace(namespace).withName(name).replace(resource);
-                }
+            if (spec != null && updateImageName(entity, spec.getTemplate(), imagePrefix, imageName)) {
+                kubernetes.apps().deployments().inNamespace(namespace).withName(name).replace(resource);
+                kubernetes.apps().deployments().inNamespace(namespace).withName(name).rolling().restart();
```
Comment (rohanKanojia) [inline jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java:150]: Umm, why is this required? As far as I remember `replace()` also triggers rolling update

Comment (manusa) [reply jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java:150]: It doesn't, at least in this version. Check the code, maybe I missed something.

**Review by rohanKanojia — APPROVED** [review]

File: jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java (hunk 1 above)
Comment (rohanKanojia) [reply jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java:150]: ohk, I tested it and looks like image doesn't get updated with a single `replace()`.

Comment (manusa) [reply jkube-kit/watcher/standard/src/main/java/org/eclipse/jkube/watcher/standard/DockerImageWatcher.java:150]: We should revisit this once we update Kubernetes Client. If it's still not working, maybe evaluate alternatives and improve Kubernetes Client for this scenario.
